nf_core/modules/lint/main_nf.py

```diff
@@ -331,8 +331,11 @@ def check_process_section(self, lines, fix_version, progress_bar):
             log.debug(f"Unable to connect to url '{urlunparse(url)}' due to error: {e}")
             self.failed.append(("container_links", "Unable to connect to container URL", self.main_nf))
             continue
-        if response.status_code != 200:
-            self.failed.append(("container_links", "Unable to connect to container URL", self.main_nf))
+        if not response.ok:
+            if response.status_code in [400, 401, 402, 403]:
+                self.warned.append(("container_links", "Access to container URL denied", self.main_nf))
+            else:
+                self.failed.append(("container_links", "Unable to connect to container URL", self.main_nf))
```

Comment:
I would add the status code to the message as you suggested, it will be easier to debug.